internal/testutil/treedump_test.go

```diff
@@ -0,0 +1,92 @@
+package testutil_test
```

Comment:
This is a good idea and a good improvement, testing the testing utilities. Do we have some existing tests for treedump? If not, I would be inclined to not include them in this PR and add them later on as an enhancement because the PR is already above 1000 lines.

Reply:
@rebornplusplus

Reply:
I would much rather have this inside this PR if this doesn't block the PR.

Good chances are we won't be adding these tests in a separate PR in the near future.

Reply:
> Do we have some existing tests for treedump?

No.

Reply:
Agreed offline: The PR is too big and this test file might grow in the future as well. Let's split this test file into a new PR in Alberto's fork and potentially include in one of the smaller PRs in the future.

Reply:
Thanks for this Rafid.